model/lifecycle.go

```diff
@@ -921,8 +867,73 @@ func createTasksForBuild(project *Project, buildVariant *BuildVariant, b *build.
 			}
 		}
 
-		// append the task to the list of the created tasks
-		tasks = append(tasks, newTask)
+		taskMap[newTask.Id] = newTask
+	}
+
+	// Create display tasks
+	tasks := task.Tasks{}
+	for _, dt := range buildVariant.DisplayTasks {
+		id := displayTable.GetId(b.BuildVariant, dt.Name)
+		if id == "" {
+			continue
+		}
+		if !createAll && !utility.StringSliceContains(displayNames, dt.Name) {
+			continue
+		}
+		execTaskIds := []string{}
+		for _, et := range dt.ExecutionTasks {
+			execTaskId := execTable.GetId(b.BuildVariant, et)
+			if execTaskId == "" {
+				grip.Error(message.Fields{
+					"message":         "execution task not found",
+					"variant":         b.BuildVariant,
+					"exec_task":       et,
+					"available_tasks": execTable,
+					"project":         project.Identifier,
+				})
+				continue
+			}
+			execTaskIds = append(execTaskIds, execTaskId)
+		}
+		newDisplayTask, err := createDisplayTask(id, dt.Name, execTaskIds, buildVariant, b, v, project, createTime)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Failed to create display task %s", id)
+		}
+		newDisplayTask.GeneratedBy = generatedBy
+
+		// add every execution task dependency onto the display task, without duplicates
+		depMap := make(map[string]bool)
+		for _, etID := range newDisplayTask.ExecutionTasks {
+			if _, ok := taskMap[etID]; ok {
+				taskMap[etID].DisplayTask = newDisplayTask
+				for _, dep := range taskMap[etID].DependsOn {
+					if !depMap[dep.TaskId] {
+						depMap[dep.TaskId] = true
+						newDisplayTask.DependsOn = append(newDisplayTask.DependsOn, dep)
+					}
+				}
+			} else {
+				execTask, err := task.FindOneId(etID)
```

Comment:
we can get just the DependsOn field here

Reply:
Again at the risk of devolving into philosophical debate: 
I'd argue in general `WithFields` is an invitation for subtle bugs and that the networking/latency advantages are outweighed by the maintainability liability of a task that's missing fields. Special cases (like fetching every task in every  build in a version, for example) make that liability worthwhile to take on.

Reply:
Do you expect this to be a special case or to happen pretty frequently? I think since it's clear this task is used only for a few lines, there's no reason to make it less efficient.

Reply:
I imagine it happening often. For example, server does this a lot to include the (existing) generator in a display task it just generated.

However, the maintainability argument is that although it's clear to us _today_ someone will add 100 lines and at the end of the 100 lines someone else will use one of these tasks not realizing the task is an empty shell.

I have an idea for an improvement/compromise: I'll batch the existing execution tasks in a single query so we don't incur query overhead more than once per display task.

Reply:
That seems reasonable--although one more plug for WithFields considering this is frequent, if you call the variable something like `execTaskDependencies` then it will be clearer to future programmers that it's not a full task.

Reply:
Your suggestion gave me an idea I think will work for both of us: I moved dependency gathering into a dedicated function where we fetch only the dependencies.